packages/sit-onyx/src/components/OnyxNavButton/OnyxNavButton.vue

```diff
@@ -0,0 +1,136 @@
+<script lang="ts" setup>
+import { isExternalLink } from "../../utils";
+import arrowSmallUpRight from "@sit-onyx/icons/arrow-small-up-right.svg?raw";
+import OnyxFlyoutMenu from "../OnyxFlyoutMenu/future/OnyxFlyoutMenu.vue";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import { type OnyxNavButtonProps } from "./types";
+import { type VNode } from "vue";
+
+const props = withDefaults(defineProps<OnyxNavButtonProps>(), {
+  active: false,
+  withExternalIcon: "auto",
+});
+
+const slots = defineSlots<{
+  /**
+   * An optional slot to override the label content (e.g. an `OnyxBadge`).
+   */
+  default?(): unknown;
+  /**
+   * An optional slot to render nested children.
+   */
+  children?(): VNode[];
+}>();
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the nav button is clicked (via click or keyboard).
+   */
+  click: [href: string];
+}>();
+
+const shouldShowExternalIcon = (args: OnyxNavButtonProps) => {
+  const withExternalIcon = args.withExternalIcon ?? "auto";
+
+  if (withExternalIcon !== "auto") return args.withExternalIcon;
+  return isExternalLink(args.href ?? "");
+};
+</script>
+
+<template>
+  <OnyxFlyoutMenu class="onyx-nav-button">
+    <button
+      class="onyx-nav-button__trigger onyx-text"
+      :class="{
+        'onyx-nav-button--active': props.active,
+      }"
+      @click="props.href && emit('click', props.href)"
+    >
+      <slot>
+        <span>{{ props.label }}</span>
+        <OnyxIcon
+          v-if="shouldShowExternalIcon(props)"
+          class="onyx-nav-button__icon"
+          :icon="arrowSmallUpRight"
+          size="16px"
+        />
+      </slot>
+    </button>
+    <template v-if="slots.children" #options>
+      <slot name="children"></slot>
+    </template>
+  </OnyxFlyoutMenu>
+</template>
+
+<style lang="scss">
+@use "../../styles/mixins/layers.scss";
+
+.onyx-nav-button {
+  @include layers.component() {
+    $gap: var(--onyx-spacing-2xs);
+
+    .onyx-future-flyout-menu__list {
```

Comment:
Shouldn't these styles be applied by the OnyxFlyoutMenu itself? Because they are the same for all flyouts (nav button, user menu etc.)